File: app/src/main/java/org/baiyu/fuckshare/ClearCacheActivity.java
package org.baiyu.fuckshare;

import android.app.Activity;
import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.Nullable;

import java.io.File;

import timber.log.Timber;

public class ClearCacheActivity extends Activity {
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        String message;

        File[] files = getCacheDir().listFiles();
        if (files == null || files.length == 0) {
            message = getResources().getString(R.string.clear_cache_empty);
            Timber.d("No cache");
        } else {
            // clear all cache
            boolean status = Utils.clearCache(this, 0);
            if (status) {
                message = getResources().getString(R.string.clear_cache_success);
            } else {
                message = getResources().getString(R.string.clear_cache_failed);
            }
            Timber.d("Cache cleared with result: %b", status);
        }
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
        finish();
    }
}

File: app/src/main/java/org/baiyu/fuckshare/ClearCacheActivity.kt
package org.baiyu.fuckshare

import android.app.Activity
import android.os.Bundle
import android.widget.Toast
import org.baiyu.fuckshare.utils.AppUtils
import timber.log.Timber

class ClearCacheActivity : Activity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val files = cacheDir.listFiles()
        if (files?.isEmpty() == true) {
            Timber.d("No cache")
            resources.getString(R.string.clear_cache_empty)
        } else {
            // clear all cache
            val status = AppUtils.clearCache(this, 0)
            Timber.d("Cache cleared with result: %b", status)
            if (status) {
                resources.getString(R.string.clear_cache_success)
            } else {
                resources.getString(R.string.clear_cache_failed)
            }
        }.also {
            Toast.makeText(this, it, Toast.LENGTH_SHORT).show()
        }
        finish()
    }
}